jetty-core/jetty-compression/jetty-compression-server/src/main/java/org/eclipse/jetty/compression/server/CompressionConfig.java

```diff
@@ -485,6 +554,81 @@ public Builder compressPathInclude(String pathSpecString)
             return this;
         }
 
+        /**
+         * Control the preferred order of encoders when compressing response content.
```

Comment:
Is it really an "order"?    If the client sends an accept-encoding header with actual values, then their order is used.
If they send "*", then we do use this as an order, but only the first element would ever be used.

Maybe this would be better with an acceptable encodings set and a default encoding single value?

Reply:
No, it is meant to be an order.

This PR covers 2 issues.
The first one is the `Accept-Encoding: *` request header.
The second is the desire for users to override the order of the `Accept-Encoding` value list (even qualified order) of the client and do things like "use zstd over gzip"

Reply:
The configuration is meant to allow the user to have a specific pathspec have different rules, such as wanting to use brotli if the user-agent supports it at all, but still fall back to gzip if the user-agent doesn't support brotli.

This order is also doubles as the order to use when `Accept-Encoding: *` is used.
If the user-agent sends `Accept-Encoding: *` and there is no such configurable, then there is no way to select the encoding to use.

Also, the `Accept-Encoding: *` has a 3rd feature, _that I need to add unit tests for_, the configuration should support having a `compressEncodingExclude("*")` to not perform response compression clients that use `Accept-Encoding: *` as an alternate setup.

Reply:
AH OK!